libsmtutil/SMTLib2Interface.cpp

```diff
@@ -212,33 +213,40 @@ std::string SMTLib2Interface::toSExpr(Expression const& _expr)
 	return sexpr;
 }
 
-std::string SMTLib2Interface::toSmtLibSort(Sort const& _sort)
+std::string SMTLib2Interface::toSmtLibSort(solidity::smtutil::SortPointer _sort)
 {
-	switch (_sort.kind)
+	if (!m_sortNames.count(_sort))
+		m_sortNames[_sort] = toSmtLibSortInternal(_sort);
+	return m_sortNames.at(_sort);
```

Comment:
What is it about `at` that you don't like?

Reply:
`at()` throws an exception if the element is not present in the container, but you already check for that in the `if` above, so it's just unnecessary overhead.

Reply:
Well, for `map` `operator []` default-initializes a new value and insert the new pair into the map, if the key was not present. So I think it has even more overhead than `at`.

Reply:
Sure, but not in this case, because you've already checked for the existence of the element. It's not a big deal regardless, so no need to make this change.